spec/controllers/admin/comments_controller_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'spec_helper'
+
+describe Admin::CommentsController, type: :controller do
+
+  # It is necessary to use bang version of let to build roles before user
+  let(:conference) { create(:conference) }
+  let!(:first_user) { create(:user) }
+  let!(:organizer_role) { create(:role, name: 'organizer', resource: conference) }
+  let(:organizer) { create(:user, role_ids: organizer_role.id, last_sign_in_at: Time.now - 1.day) }
+  let(:participant) { create(:user) }
+  let(:event) { create(:event, conference: conference) }
+  let(:comment) { create(:comment, commentable_type: 'Event', commentable_id: event.id) }
+
+  context 'not logged in user' do
+    describe 'GET #index' do
+      it 'renders the :index template' do
+        comment
+        get :index
+        expect(response).to redirect_to(user_session_path)
```

Comment:
there is no flash when a non-logged user tries to access protected area, but it is a redirect to login page